Feat/specialized freelancer search hook

yhoungdev opened the pull request:
# 📝 Pull Request

## 🛠️ Issue
Closes #issue-ID

## 📚 Description
Implemented a freelancer search system with advanced filtering, caching, and performance optimizations.

NB: i added some JsDoc for proper typing

## ✅ Changes
- Search Hook: skills, location, price, rating, availability   
- Search History: persistent favorites & quick access  
- Filters: multi-criteria state management with validation  
- Performance: debouncing, pagination, optimization

KevinMB0220 commented:
> Hey @Josue19-08 you can take a look at this

Please fix the ci first

yhoungdev commented:
Ok, i would implement all the coderabbit fixe suggestions

KevinMB0220 commented:
> Ok, i would implement all the coderabbit fixe suggestions

Yeah thanks

yhoungdev commented on src/hooks/use-search-cache.ts:
```diff
@@ -0,0 +1,251 @@
+"use client"
+
+import { useCallback, useRef, useMemo } from 'react'
```
Adding that would have unused items

yhoungdev commented on src/hooks/use-search-cache.ts:
```diff
@@ -0,0 +1,251 @@
+
+  const cacheRef = useRef<SearchCache>(new SearchCache(maxSize, defaultTTL))
+  const cleanupIntervalRef = useRef<NodeJS.Timeout | null>(null)
+  const isEnabledRef = useRef<boolean>(true)
+
```
This would break the app

yhoungdev commented on src/utils/search-helpers.ts:
```diff
@@ -0,0 +1,503 @@
+      priceFilter.currency
+    )
+    
+    const meetsMin = !priceFilter.min || freelancerRate >= priceFilter.min
+    const meetsMax = !priceFilter.max || freelancerRate <= priceFilter.max
+    
```
Fixed

yhoungdev commented:
@KevinMB0220 i have updated the items, you can check now.

KevinMB0220 commented:
> @KevinMB0220 i have updated the items, you can check now.

Yeah looks good all, the problems its only the ci

yhoungdev commented:
Yea sure.

KevinMB0220 commented:
> Updated codebase.

Yeah you have liting error with the ci

yhoungdev commented:
Funny how vercel log is trowing same error, even when it has been resolved

i resolved with removing unused error callback with 

``` } catch {
      return timezone.split('/').pop()?.replace(/_/g, ' ') || timezone
    }```

KevinMB0220 commented:
> Funny how vercel log is trowing same error, even when it has been resolved
> 
> i resolved with removing unused error callback with
> 
> ```
>       return timezone.split('/').pop()?.replace(/_/g, ' ') || timezone
>     }```
> ```

ok do it

yhoungdev commented:
I removed it already.

KevinMB0220 commented:
> I removed it already.

Run this so you can see the errors 

npm run lint

yhoungdev commented:
> > I removed it already.
> 
> Run this so you can see the errors
> 
> npm run lint

Yeah, I ran it and fixed the `ERROR```

There are only warnings throughout the codebase now 



https://github.com/user-attachments/assets/b35d22bb-84bd-4f17-822b-a7c8f51587a7



And also note, I made another PR https://github.com/OFFER-HUB/offer-hub/issues/678 and it has the latest update on the cli error mentions

KevinMB0220 commented:
> Yes, it's not an error, per say. My theory is that Vercel is treating warning messages as errors. 


Yeah maybe, I merge one and see what it's happening
